src/resolve-dependency.ts

```diff
@@ -1,127 +1,84 @@
-  ) {
-    const trailingSlash = specifier.endsWith("/");
-    resolved = await resolvePath(
-      resolve(parent, "..", specifier) + (trailingSlash ? "/" : ""),
-      parent,
-      job
-    );
-  } else if (specifier[0] === "#") {
+  if (isAbsolute(specifier) || specifier === '.' || specifier === '..' || specifier.startsWith('./') || specifier.startsWith('../')) {
+    const trailingSlash = specifier.endsWith('/');
+    resolved = await resolvePath(resolve(parent, '..', specifier) + (trailingSlash ? '/' : ''), parent, job);
+  }
+  else if (specifier[0] === '#') {
     resolved = await packageImportsResolve(specifier, parent, job, cjsResolve);
-  } else {
+  }
+  else {
     resolved = await resolvePackage(specifier, parent, job, cjsResolve);
   }
 
   if (Array.isArray(resolved)) {
-    return Promise.all(
-      resolved.map((resolved) => job.realpath(resolved, parent))
-    );
-  } else if (resolved.startsWith("node:")) {
+    return Promise.all(resolved.map(resolved => job.realpath(resolved, parent)))
+  } else if (resolved.startsWith('node:')) {
     return resolved;
   } else {
     return job.realpath(resolved, parent);
   }
-}
+};
 
-async function resolvePath(
-  path: string,
-  parent: string,
-  job: Job
-): Promise<string> {
-  const result =
-    (await resolveFile(path, parent, job)) ||
-    (await resolveDir(path, parent, job));
+async function resolvePath (path: string, parent: string, job: Job): Promise<string> {
+  const result = await resolveFile(path, parent, job) || await resolveDir(path, parent, job);
   if (!result) {
     throw new NotFoundError(path, parent);
   }
   return result;
 }
 
-async function resolveFile(
-  path: string,
-  parent: string,
-  job: Job
-): Promise<string | undefined> {
-  if (path.endsWith("/")) return undefined;
+async function resolveFile (path: string, parent: string, job: Job): Promise<string | undefined> {
+  if (path.endsWith('/')) return undefined;
   path = await job.realpath(path, parent);
   if (await job.isFile(path)) return path;
-  if (
-    job.ts &&
-    path.startsWith(job.base) &&
-    path.slice(job.base.length).indexOf(sep + "node_modules" + sep) === -1 &&
-    (await job.isFile(path + ".ts"))
-  )
-    return path + ".ts";
-  if (
-    job.ts &&
-    path.startsWith(job.base) &&
-    path.slice(job.base.length).indexOf(sep + "node_modules" + sep) === -1 &&
-    (await job.isFile(path + ".tsx"))
-  )
-    return path + ".tsx";
-  if (await job.isFile(path + ".js")) return path + ".js";
-  if (await job.isFile(path + ".json")) return path + ".json";
-  if (await job.isFile(path + ".node")) return path + ".node";
+  if (job.ts && path.startsWith(job.base) && path.slice(job.base.length).indexOf(sep + 'node_modules' + sep) === -1 && await job.isFile(path + '.ts')) return path + '.ts';
+  if (job.ts && path.startsWith(job.base) && path.slice(job.base.length).indexOf(sep + 'node_modules' + sep) === -1 && await job.isFile(path + '.tsx')) return path + '.tsx';
```

Comment:
Perhaps abstract this into an `extensions` configuration rather, so it can be configured and not necessarily the default (sincethis will match `./file.ts` before `file.js`).